plugins/volume/volumeset.go

```diff
@@ -0,0 +1,86 @@
+package volume
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/plugins"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+type volume interface {
+	buildSandbox(engines.SandboxBuilder) error
+	dispose() error
+	String() string
+}
+
+type volumeOptions struct {
+	taskID     string
+	runID      int
+	name       string
+	mountPoint string
+}
+
+// volumeSet acts as a single task plugin, but is a wrapper around each of the volume
+// plugins that were initialized in response to the task payload.
+// This plugin will response to typical task plugin events, such as Prepare or Dispose,
+// and in turn will call the appropriate methods on each volume plugin for that stage.
+type volumeSet struct {
+	plugins.TaskPluginBase
+	// Group of volumes that should be managed as one during the task life cycle
+	volumes []volume
+	context *runtime.TaskContext
+	payload payload
+	vm      *volumeManager
+}
+
+func (vs *volumeSet) Prepare(context *runtime.TaskContext) error {
+	vs.context = context
+
+	for volumeType, plugin := range vs.vm.plugins {
+		switch volumeType {
+		case "persistent":
+			for _, vol := range vs.payload.Persistent {
+				opts := &volumeOptions{
+					taskID:     context.TaskInfo.TaskID,
+					runID:      context.TaskInfo.RunID,
+					name:       vol.Name,
+					mountPoint: vol.MountPoint,
+				}
+				v, err := plugin.newVolume(opts)
+				if err != nil {
+					return err
+				}
+				vs.volumes = append(vs.volumes, v)
+			}
+		}
```

Comment:
Do you think we should still handle if the volume type doesn't exist?  I don't think we should panic since that would cause the worker to crash if someone submitted a bad task payload, but we could log a task error.